classical_algorithms/classic_algorithm_cpp/quick_sort.cpp

```diff
@@ -0,0 +1,56 @@
+#include<iostream>
+using namespace std;
+
+void quick_sort(int[],int,int);
+
+int partition(int[],int,int);
+
+void quick_sort(int a[],int start,int end)
+{
+    if(start>=end)
+        return;
+    int pindex=partition(a,start,end);
+    quick_sort(a,start,pindex-1);
+    quick_sort(a,pindex+1,end);
+ 
+}
+
+int partition(int a[],int start,int end)
+{
+    int pivot,pindex,temp;
+    pivot=a[end];
+    pindex=start;
+    for(int i=start;i<=end-1;i++)
+        if(a[i]<=pivot)
+        {
+          temp=a[i];
+          a[i]=a[pindex];
+          a[pindex]=temp;
+          pindex=pindex+1;
+        }
+    temp=a[pindex];
+    a[pindex]=a[end];
+    a[end]=temp;
+    return pindex;
+}
+int main()
+{
+int n;
+  int a[n] = {0};
```

Comment:
initializing an array with size n before giving n gives an error. I will resolve this issue in my further commits